File: loan-management-api/src/modules/payments/payment.controller.ts
import { Request,Response } from "express";
import { PaymentService } from "./payment.service";

export class PaymentController{
    
    constructor(private paymentService:PaymentService){}

    async getPayments(req:Request,res:Response){
      const payments=await this.paymentService.getPayments();
      res.status(200).json(payments);
    }

    async createPayment(req:Request,res:Response){
        const payment=await this.paymentService.createPayment(req.body);
        res.status(201).json({msg:"Pago creado con éxito",payment});
    }

    async deletePayment(req:Request,res:Response){
        await this.paymentService.deletePayment(Number(req.params.id));
        res.status(200).json({msg:"Pago eliminado correctamente!"});
    }

}

File: springapi/src/main/java/sisPrestamo/Brayan/modules/payments/PaymentController.java
package sisPrestamo.Brayan.modules.payments;

import java.util.Map;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.ModelAttribute;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import jakarta.validation.Valid;
import sisPrestamo.Brayan.modules.payments.dtos.PaymentRequest;
import sisPrestamo.Brayan.modules.payments.dtos.PaymentResponse;
import sisPrestamo.Brayan.shared.ResponseBuilder;

@RestController
@RequestMapping("/api/payments")
public class PaymentController {
    private final PaymentService paymentService;

    public PaymentController(PaymentService paymentService){
this.paymentService=paymentService;
    }

    @GetMapping
    public ResponseEntity<Map<String,Object>> getPayments(){
        
        return ResponseEntity.ok(new ResponseBuilder().add("payments", paymentService.getPayments()).build());
    }

    @PostMapping
    public ResponseEntity<Map<String,Object>> createPayment(@Valid @ModelAttribute PaymentRequest payment){
         PaymentResponse paymentCreate=paymentService.createPayment(payment);
         return ResponseEntity.status(HttpStatus.CREATED).body(new ResponseBuilder().msg("Pago creado correctamente").add("payment", paymentCreate).build());
    }
    
}
